### PR TITLE
Asset hashing with tests

### PR DESCRIPTION
**Pull request by geek-caroline**
This adds in the ability to hash page asset filenames (and includes tests for this functionality)

### CODE REVIEW

File: .jshintrc [hunk 1]
```diff
@@ -12,5 +12,6 @@
     "immed": true,
     "trailing": true,
     "smarttabs": true,
-    "node": true
+    "node": true,
+    "predef": ["describe", "it", "before", "after", "afterEach"]
```
Comment (AlbertoElias) [inline .jshintrc]: This is not needed as they are set in each test file

File: lib/tasks/build.js [hunk 2]
```diff
@@ -65,7 +72,12 @@ module.exports.js = function(gulp, config) {
 		return bundle.bundle(config)
 			.pipe(source(dest))
 			.pipe(gulpif(config.env === 'production', streamify(uglify())))
+			.pipe(gulpif(config.hash, streamify(hash(hashOptions))))
+			.pipe(gulpif(config.hash, gulp.dest(destFolder)))
+			.pipe(gulpif(config.hash, hash.manifest(dest + '-asset-hash.json')))
 			.pipe(gulp.dest(destFolder));
+	} else {
+		throw Error('Invalid path to main JavaScript file');
```
Comment (AlbertoElias) [inline lib/tasks/build.js]: This has been removed, same as below

File: test/tasks/build.test.js [hunk 3]
```diff
@@ -110,6 +110,37 @@ describe('Build task', function() {
...
+		it('should build a hashed version of the js', function(done) {
+			build
+				.js(gulp, {
+					hash: true
+				})
+				.on('end', function() {
+					var builtJsJson = fs.readFileSync('build/main.js-asset-hash.json', 'utf8');
+					expect(builtJsJson.indexOf('main.js')).to.not.be(-1);
+					var jsFileName = JSON.parse(builtJsJson)['main.js'],
+						jsFileContents = fs.readFileSync('build/' + jsFileName, 'utf8');
```
Comment (AlbertoElias) [inline test/tasks/build.test.js]: Indentation

File: test/tasks/build.test.js [hunk 4]
```diff
@@ -199,5 +230,38 @@ describe('Build task', function() {
...
+		it('should build a hashed version of the css', function(done) {
+			build
+				.sass(gulp, {
+					hash: true
+				})
+				.on('end', function() {
+					var builtCssJson = fs.readFileSync('build/main.css-asset-hash.json', 'utf8');
+					expect(builtCssJson.indexOf('main.css')).to.not.be(-1);
+					var cssFileName = JSON.parse(builtCssJson)['main.css'],
+						cssFileContents = fs.readFileSync('build/' + cssFileName, 'utf8');
```
Comment (AlbertoElias) [inline test/tasks/build.test.js]: Indentation. We prefer to use var twice

File: .jshintrc (hunk 1 above)
Comment (geek-caroline) [inline .jshintrc]: Great, don't think they originally were when I started merging in the test framework - I shall remove

**Pull request merged**